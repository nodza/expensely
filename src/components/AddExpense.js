import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';


export class AddExpense extends Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push('/dashboard');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="container">
            <h1 className="page-header__title">
              Add Expense
            </h1>
          </div>
        </div>
        <div className="container">
          <ExpenseForm
            onSubmit={this.onSubmit}
          />
        </div>
        
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpense);