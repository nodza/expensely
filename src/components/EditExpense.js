import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import EditExpenseForm from './ExpenseForm';

export class EditExpense extends Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/dashboard');
  };

  onRemove = () => {
    this.props.startRemoveExpense({id: this.props.expense.id });
    this.props.history.push('/dashboard');
  };


  render() {
    return (
      <div>
        <EditExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit} />
        <br />
        <button onClick={this.onRemove}>Delete Expense</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch, props) => {
  return {
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);