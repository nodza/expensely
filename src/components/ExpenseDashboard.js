import React from 'react';

const ExpenseDashboard = () => (
  <div>Hello from the ExpenseDashboard</div>
);

export default ExpenseDashboard;