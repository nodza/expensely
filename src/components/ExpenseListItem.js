import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

// onRemoveExpense = () => {
//   console.log("Removal in process...");
// };

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
  <div>
    <h4>{description}</h4>
    <p>{amount} - {createdAt}</p>
    <button onClick={()  => {
      dispatch(removeExpense({ id }));
    }}>Remove</button>
  </div>
);

export default (connect())(ExpenseListItem);