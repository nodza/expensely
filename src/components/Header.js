import React from 'react';
import { NavLink} from 'react-router-dom';

const Header = () => (
  <header>
    <h2>Expense.ly</h2>
    <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Add New</NavLink>
    <NavLink to="/help" activeClassName="is-active">Help</NavLink>
  </header>
);

export default Header;