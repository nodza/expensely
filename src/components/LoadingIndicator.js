import React from 'react';

const LoadingIndicator = () => (
  <div className="loader">
    <img className="loader__image" src="/images/loader.svg" alt="Loading..." />
  </div>
);

export default LoadingIndicator;