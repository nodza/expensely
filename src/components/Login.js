import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const Login = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h2 className="box-layout__title">ExpenseIt</h2>
      <p>Sign in to get started</p>
      <button className="button" onClick={startLogin}>Sign in with Google</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(Login);