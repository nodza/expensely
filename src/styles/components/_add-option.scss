.add-option-error {
  color: $warning-red;
  font-size: $medium-size;
  margin: 0;
  text-align: center;
  padding: 1.2rem;
  background: #ffe8ec;
}

.add-option {
  display: flex;
  padding: $small-size;
  flex-direction: column;
}

.add-option__input {
  flex-grow: 1;
  border: none;
  background: $light-bg;
  margin: 0 0 $small-size 0;
  padding: $small-size;
}

@media (min-width: $desktop-breakpoint) {
  .add-option {
    flex-direction: row;
  }

  .add-option__input {
    margin: 0 $small-size 0 0;
  }
}
