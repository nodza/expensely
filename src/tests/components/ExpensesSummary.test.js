import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary component with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={335} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary component with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={32} expensesTotal={9877620} />);
  expect(wrapper).toMatchSnapshot();
});