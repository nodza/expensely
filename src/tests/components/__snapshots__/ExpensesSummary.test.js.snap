// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary component with 1 expense 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling 
  $3.35
  .
</div>
`;

exports[`should render ExpensesSummary component with multiple expenses 1`] = `
<div>
  Viewing 
  32
   
  expenses
   totalling 
  $98,776.20
  .
</div>
`;
